client/gui/ShortcutHandler.cpp

```diff
@@ -0,0 +1,362 @@
+/*
+ * ShortcutHandler.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+
+#include "StdInc.h"
+
+#include "ShortcutHandler.h"
+#include "Shortcut.h"
+#include <SDL_keycode.h>
+
+std::vector<EShortcut> ShortcutHandler::translateKeycode(SDL_Keycode key) const
+{
+	static const std::multimap<SDL_Keycode, EShortcut> keyToShortcut = {
```

Comment:
As adding/modifying/removing new shortcut requires modifications in 3 such places (EShortcut, keyToShortcut and shortcutNames below) it's better to keep all of them in single file.
Consider moving keyToShortcut and shortcutNames to Shortcut.h

Reply:
Not 3 places.
At the very least you also need to add actual usage of shortcut, in UI code of window that will use said shortcut and likely - button with assigned shortcut in widget configuration json.

So even if I do this, you still won't be able to define shortcuts in one place.

Also want to avoid placing unnecessary info in header, to avoid increase in build times & to keep API minimalistic.

And don't see a way to force compile-time check to detect invalid shortcuts - since it is legal for a shortcut to not have a name and assigned key.